src/crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node.h

```diff
@@ -0,0 +1,147 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+
+#pragma once
+
+#include <boost/iterator/counting_iterator.hpp>
+
+#include "crimson/common/log.h"
+#include "crimson/os/seastore/seastore_types.h"
+#include "crimson/os/seastore/transaction_manager.h"
+#include "crimson/os/seastore/extentmap_manager.h"
+
+namespace crimson::os::seastore::extentmap_manager{
+
+struct extmap_node_meta_t {
+  depth_t depth = 0;
+
+  std::pair<extmap_node_meta_t, extmap_node_meta_t> split_into(objaddr_t pivot) const {
+    return std::make_pair(
+      extmap_node_meta_t{depth},
+      extmap_node_meta_t{depth});
+  }
+
+  static extmap_node_meta_t merge_from(
+    const extmap_node_meta_t &lhs, const extmap_node_meta_t &rhs) {
+    assert(lhs.depth == rhs.depth);
+    return extmap_node_meta_t{lhs.depth};
+  }
+
+  static std::pair<extmap_node_meta_t, extmap_node_meta_t>
+  rebalance(const extmap_node_meta_t &lhs, const extmap_node_meta_t &rhs, laddr_t pivot) {
+    assert(lhs.depth == rhs.depth);
+    return std::make_pair(
+      extmap_node_meta_t{lhs.depth},
+      extmap_node_meta_t{lhs.depth});
+  }
+};
+
+struct ExtMapNode : LogicalCachedExtent {
+  using ExtMapNodeRef = TCachedExtentRef<ExtMapNode>;
+
+  ExtMapNode(ceph::bufferptr &&ptr) : LogicalCachedExtent(std::move(ptr)) {}
+  ExtMapNode(const ExtMapNode &other)
+  : LogicalCachedExtent(other) {}
+
+  LogicalCachedExtentRef make_duplicate(TransactionManager &tm, Transaction& t,
+	  LogicalCachedExtentRef ref) {
+    return tm.get_mutable_extent(t, ref);
+  };
+
+  using find_lextent_ertr = ExtentMapManager::find_lextent_ertr;
+  using find_lextent_ret = ExtentMapManager::find_lextent_ret;
+  virtual find_lextent_ret find_lextent(TransactionManager &tm, Transaction &t,
```

Comment:
I think you should go ahead and create a context struct with tm an t as I did in the btree lba manager.  It'll make a lot of things easier to read.

Reply:
updated to use extmap_context_t to include transactionmanager and transaction.